### PR TITLE
fix docker build

### PR DESCRIPTION
**Pull request by jackalcooper**

### CODE REVIEW

**Comment by jackalcooper** [comment]
@yuanming-hu  Python3.6 is now installed from source. That's what i got after running `ti test`:
```
Traceback (most recent call last):
  File "/app/taichi/bin/ti", line 3, in <module>
    import taichi
  File "/app/taichi/python/taichi/__init__.py", line 1, in <module>
    from taichi.main import main
  File "/app/taichi/python/taichi/main.py", line 6, in <module>
    from taichi.tools.video import make_video, interpolate_frames, mp4_to_gif, scale_video, crop_video, accelerate_video
  File "/app/taichi/python/taichi/tools/video.py", line 3, in <module>
    import taichi.core as core
  File "/app/taichi/python/taichi/core/__init__.py", line 1, in <module>
    from .util import tc_core, build, format, load_module, start_memory_monitoring, \
  File "/app/taichi/python/taichi/core/util.py", line 156, in <module>
    os.symlink(link_src, link_dst)
FileNotFoundError: [Errno 2] No such file or directory: '/app/taichi/python/taichi/core/../lib/taichi_core.so' -> '/app/taichi/python/taichi/core/../lib/l
ibtaichi_core.so' 
```

**Comment by jackalcooper** [comment]
The .so file is in build directory
```
root@b6fe8da8aadb:/app/taichi# ls
CMakeLists.txt  Jenkinsfile  README.md     benchmarks  build        cmake  examples  misc    taichi
Dockerfile      LICENSE      appveyor.yml  bin         ci_setup.py  docs   external  python  tests
root@b6fe8da8aadb:/app/taichi# cd build/
root@b6fe8da8aadb:/app/taichi/build# ls
CMakeCache.txt  CMakeFiles  Makefile  cmake_install.cmake  libtaichi_core.so
```

File: Dockerfile [hunk 1]
```diff
@@ -60,9 +59,17 @@ WORKDIR /app
...
-RUN cd /app/taichi/build && make -j `nproc --all`
+RUN mkdir /app/taichi/build
+WORKDIR /app/taichi/build
+RUN export CUDA_BIN_PATH=/usr/local/cuda-10.0 && \
+    cmake .. -DPYTHON_EXECUTABLE=$(which python3.6) -DCUDA_VERSION=10.0 -DTLANG_WITH_CUDA:BOOL="True" && \
+    make -j 15 && \
+    ldd libtaichi_core.so
+
+WORKDIR /app/taichi/python
+ENV PATH="/app/taichi/bin:$PATH"
```
Comment (yuanming-hu) [inline Dockerfile]: Please add after this line:
```
ENV TAICHI_REPO_DIR="/app/taichi"
```
This will probably solve your `libtaichi_core.so` path issue.

**Comment by jackalcooper** [comment]
after fixing the env var (which I should have noticed sorry) it got stuck here:
```
Step 34/36 : RUN ti test
 ---> Running in 84663a3b0509
[T 01/31/20 01:08:05.909] [logging.cpp:Logger@68] Taichi core started. Thread ID = 7
Making output directory
[Taichi version 0.3.25, cuda 10.0, commit 43494f8f]

 *******************************************
 **     Taichi Programming Language       **
 *******************************************


Running python tests...

============================= test session starts ==============================
platform linux -- Python 3.6.0, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /app/taichi
collected 187 items

../tests/python/test_abs.py .                                            [  0%]
../tests/python/test_ad_atomic.py 
```

**Comment by yuanming-hu** [comment]
Great! It seems to me that the tests are running - usually, it takes 5 minutes to finish. Is it making any progress?

**Comment by jackalcooper** [comment]
Zero progress so far. It's been half a hour at least.

**Comment by jackalcooper** [comment]
Does this mean that the first test for abs is passed and it got stuck when running test case for  atomic add?
```
collected 187 items

../tests/python/test_abs.py .                                            [  0%]
../tests/python/test_ad_atomic.py 
```

**Comment by yuanming-hu** [comment]
Thanks for your efforts! Could you please try to build without CUDA? (set `-DTLANG_WITH_CUDA:BOOL="False"` here https://github.com/taichi-dev/taichi/pull/410/files/6555a389a67d6b49dd10ae595d728f157d344a9c#diff-3254677a7917c6c01f55212f86c57fbfR65) I feel like CUDA in docker tends to be unstable and might be the source of test stuck, yet I need your help to confirm/disconfirm.

**Comment by jackalcooper** [comment]
All tests passed if build taichi without CUDA.
```
../tests/python/test_tensor_reflection.py ..                             [ 92%]
../tests/python/test_threading.py .                                      [ 93%]
../tests/python/test_torch_ad.py ..                                      [ 94%]
../tests/python/test_torch_io.py .........                               [ 98%]
../tests/python/test_while.py ..                                         [100%]

======================= 187 passed in 189.91s (0:03:09) ========================            
```

**Comment by yuanming-hu** [comment]
Awesome! Thank you so much for your great work! I would merge this in for now, and open an issue for the CUDA problem. My feeling is that the CUDA issue is hard to resolve, as creating a virtual environment for GPU drivers sound like a tricky task and I'm not surprised if CUDA within the container is not as stable as outside the container...

**Comment by yuanming-hu** [comment]
Moving discussions here: https://github.com/taichi-dev/taichi/issues/416

**Pull request merged**